src/notification-banner/private/useNotificationBanners.jsx

```diff
@@ -0,0 +1,190 @@
+
+      if (updateBannerState.current) {
+        updateBannerState.current({ ...registeredBanners.current });
+      }
+    };
+
+    return {
+      registerNotificationBanner,
+      unregisterNotificationBanner,
+    };
+  }, []);
+
+  /**
+   * Provides the Banner Registration Context to its children.
+   */
+  const NotificationBannerProvider = ({ children }) => (
+    <BannerRegistrationContext.Provider value={bannerProviderValue}>
+      {children}
+    </BannerRegistrationContext.Provider>
+  );
+
+  NotificationBannerProvider.propTypes = {
+    children: PropTypes.node,
+  };
+
+  /**
+   * Renders a list of prioritized notification banners.
+   */
+  const NotificationBanners = () => {
+    const theme = React.useContext(ThemeContext);
+    const [banners, setBanners] = React.useState([]);
+    updateBannerState.current = setBanners;
+
+    if (!Object.keys(banners).length) {
+      return null;
+    }
+    const prioritizedBanners = organizeBannersByPriority(banners, theme.name);
+
+    return (
+      <div aria-live="polite">
+        {prioritizedBanners.map((bannerProps) => {
+          const {
+            bannerAction, custom, description, key, onRequestClose, variant,
+          } = bannerProps;
+
+          let alertType;
+          switch (variant) {
+            case 'hazard-high':
+              alertType = 'alert';
+              break;
+            case 'hazard-medium':
+              alertType = 'warning';
+              break;
+            case 'hazard-low':
+              alertType = 'info';
+              break;
+            default:
+              alertType = variant;
+          }
+
+          let actionButton = null;
+          if (bannerAction) {
+            actionButton = (
+              <Button
+                text={bannerAction.text}
+                variant="ghost"
+                data-terra-application-notification-banner={variant}
+                onClick={bannerAction.onClick}
+              />
+            );
+          }
+
+          let customIcon;
+          let customSignalWord;
+          if (alertType === 'custom' && custom !== undefined) {
+            customSignalWord = custom?.signalWord;
+
+            if (custom.customIconClass) {
+              customIcon = (
+                <svg className={cx(['custom-icon', custom.customIconClass])} />
+              );
+            }
+          }
+
+          return (
+            <Alert
+              key={key}
+              action={actionButton}
+              onDismiss={onRequestClose}
+              type={alertType}
+              customIcon={customIcon}
+              title={customSignalWord}
+              data-terra-application-notification-banner={variant}
+            >
+              {description}
+            </Alert>
+          );
+        })}
+      </div>
+    );
+  };
+
+  const useNotificationBannerExports = React.useRef({
+    NotificationBannerProvider,
+    NotificationBanners,
+  });
```

Comment:
With this implementation, we're generating new NotificationBannerProvider and NotificationBanners with every update, but only ever returning the first ones that get generated. It's a minor waste of computation time, but you could wrap this creation in a useMemo block to limit this.